File: skin-protector/src/services/commentService.js
import api from "../api";

// 특정 게시글의 댓글 목록 조회
export const getComments = async (postId) => {
    return api.get(`/comment/list/${postId}`);
};

// 댓글 작성
export const addComment = async (comment) => {
    return api.post("/comment/create", comment);
};

// 댓글 수정 (작성자만 가능)
export const updateComment = async (id, comment) => {
    return api.put(`/comment/update/${id}`, comment);
};

// 댓글 삭제 (작성자만 가능)
export const deleteComment = async (id, userId) => {
    return api.delete(`/comment/delete/${id}/${userId}`);
};

File: server_member/src/main/java/org/zerock/server_member/service/CommentService.java
package org.zerock.server_member.service;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.zerock.server_member.model.Comment;
import org.zerock.server_member.model.Member;
import org.zerock.server_member.repository.CommentRepository;
import org.zerock.server_member.repository.MemberRepository;

import java.util.List;
import java.util.Optional;

@Service
@RequiredArgsConstructor

public class CommentService {
    private final CommentRepository commentRepository;
    private final MemberRepository memberRepository;

    public List<Comment> getCommentsByPost(Long postId) {
        return commentRepository.findByPostId(postId);
    }

    public Comment addComment(Comment comment) {
        Member member = memberRepository.findById(comment.getAuthorId()).orElse(null);
        if (member != null) {
            comment.setAuthor(member.getName());
        }
        return commentRepository.save(comment);
    }

    public boolean updateComment(Long id, Comment updatedComment) {
        Optional<Comment> existingComment = commentRepository.findById(id);

        if (existingComment.isPresent()) {
            Comment comment = existingComment.get();
            if (!comment.getAuthorId().equals(updatedComment.getAuthorId())) {
                return false; // 수정 권한 없음
            }
            comment.setContent(updatedComment.getContent());
            commentRepository.save(comment);
            return true;
        }
        return false;
    }

    public boolean deleteComment(Long id, Long userId) {
        Optional<Comment> existingComment = commentRepository.findById(id);

        if (existingComment.isPresent()) {
            Comment comment = existingComment.get();
            if (!comment.getAuthorId().equals(userId)) {
                return false; // 삭제 권한 없음
            }
            commentRepository.deleteById(id);
            return true;
        }
        return false;
    }
}
